src/Rules/Validation/ScopeFormRequestValidateMethods.php

```diff
@@ -0,0 +1,72 @@
+<?php declare(strict_types=1);
+
+namespace Hihaho\PhpstanRules\Rules\Validation;
+
+use Illuminate\Foundation\Http\FormRequest;
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\ShouldNotHappenException;
+use ReflectionClass;
+use ReflectionException;
+
+final class ScopeFormRequestValidateMethods extends ScopeValidationMethods
+{
+    public function getNodeType(): string
+    {
+        return MethodCall::class;
+    }
+
+    /**
+     * @phpstan-param MethodCall $node
+     * @throws ReflectionException
+     * @throws ShouldNotHappenException
+     */
+    public function processNode(Node $node, Scope $scope): array
+    {
+        if (str_starts_with($scope->getNamespace(), 'App\\Http\\Request')) {
+            return [];
+        }
+
+        if (! $this->hasFormRequestClass($scope)) {
+            return [];
+        }
+
+        if ($this->usesValidMethod(varName: $this->nameFrom($node->var), methodName: $this->nameFrom($node))) {
+            return [];
+        }
+
+        return [
+            RuleErrorBuilder::message(
+                'Usage of unvalidated request data is not allowed outside of App\\Http\\Requests'
+            )
+                ->nonIgnorable()
+                ->tip('Use $request->safe() to use request data')
+                ->identifier('hihaho.request.unsafeRequestData')
+                ->build(),
+        ];
+    }
+
+    /** @throws ReflectionException */
+    private function hasFormRequestClass(Scope $scope): bool
```

Comment:
this loops through the methods is the inspected class and checks if the arguments are an instance/extend from the FormRequest class